File: web/types/validation.ts
export interface ValidationChecks {
  valid_format: boolean;
  valid_domain: boolean;
  can_receive_email: boolean;
  is_disposable: boolean;
  is_generic: boolean;
}

export interface ValidationAttributes {
  username: string;
  domain: string;
  is_free: boolean;
  is_disposable: boolean;
  is_generic: boolean;
  provider?: string;
  mx_record?: string;
}

export interface ValidationDetails {
  mx_found?: boolean;
  a_found?: boolean;
  mx_host?: string;
  message?: string;
}

export interface ValidationResult {
  email: string;
  status: "deliverable" | "undeliverable" | "risky" | "unknown";
  reason?: string;
  details: ValidationDetails;
  checks: ValidationChecks;
  attributes: ValidationAttributes;
  quality_score: number;
}

File: api/src/scrub/models/validation.py
from pydantic import BaseModel, EmailStr
from typing import Optional, Dict, Any


class ValidationChecks(BaseModel):
    valid_format: bool
    valid_domain: bool
    can_receive_email: bool
    is_disposable: bool
    is_generic: bool


class ValidationAttributes(BaseModel):
    username: str
    domain: str
    is_free: bool
    is_disposable: bool
    is_generic: bool
    provider: Optional[str] = None
    mx_record: Optional[str] = None


class ValidationResponse(BaseModel):
    email: EmailStr
    status: str
    reason: Optional[str] = None
    details: Dict[str, Any]
    checks: ValidationChecks
    attributes: ValidationAttributes
    quality_score: int
